src/tox/Network/Tox.lhs

```diff
@@ -26,10 +26,76 @@ lengths (e.g.  \texttt{4 | 16}), or an inclusive range (e.g. \texttt{[0,
 100]}).  Open ranges are denoted \texttt{[n,]} to mean a minimum length of
 \texttt{n} with no specified maximum length.
 
+\section{TODO: Goals and threat model}
+
+This section should give an idea on what are the goals and non-goals of Tox, so
+that reader
+
+\begin{itemize}
+  \item understands what problems Tox intends to solve
+  \item can validate if they are addresed by this specification
+  \item can make better tradeoffs and decisions in his own reimplementation of the
+    protocol
+\end{itemize}
+
+(TODO: this is just a placeholder; some more technical description should be
+given)
+
+What Tox Does:
+
+\begin{itemize}
+  \item Authentication: user's address is its public key, thus "adding friend"
+    actually means verifying key (false in case of toxme?); drawback is that
+    after being compromised you have to generate absolutely new identity; also
+    it complicates working with multiple devices; (TODO: how about some web of
+    trust, master keys and subkeys here?)
+
+  \item End-to-end encryption: all the messages are encrypted via keys derived via
+    DH, thus keys are only known to sender and receiver and are never
+    transfered over network
+
+  \item Forward secrecy (?): can be achieved by using ephemeral keys (TODO: how are
+    they used in the current protocol? is the problem actually solved?)
```

Comment:
somewhat: The specific use case that the original author planned to use toxcore personally does effectively guarantee forward secrecy. While in actuality, there's no such promise made, merely; it'll probably work.